src/amo/components/VPNPromoBanner/index.js

```diff
@@ -98,6 +115,9 @@ export class VPNPromoBannerBase extends React.Component<InternalProps> {
       label: String(impressionCount),
     });
     _localStorage.removeItem(IMPRESSION_COUNT_KEY);
+    // See https://github.com/mozilla/addons-frontend/issues/10770
+    this._updateVariant(NOT_IN_EXPERIMENT);
```

Comment:
I confirmed that we could use these few lines here instead of changing `withExperiment()`:

```js
const experiments = cookies.get(EXPERIMENT_COOKIE_NAME);
experiments[this.props.experimentId] = NOT_IN_EXPERIMENT;
cookies.set(EXPERIMENT_COOKIE_NAME, experiments, defaultCookieConfig);
```

`EXPERIMENT_COOKIE_NAME` and `defaultCookieConfig` are imported from `amo/withExperiment`. We can use `defaultCookieConfig` because the experiment config does not contain a `cookieConfig` (I checked `src/amo/experiments/20210714_amo_vpn_promo.js`).

Reply:
I commented about this above, and I'd rather not leak the internals of `withExperiment` into this component. We'll also need to add the `withCookies` HOC to this component if we want to do this, which complicates it. It also means that if we need to use this hack in the future, we'll need to add this code yet again, plus any associated tests. Keeping it in `withExperiment` keeps is encapsulated and also means that we have a permanent test in place for it. My intention would be to remove the function and hack from `withExperiment` after we have figured out a better solution to the problem.